datadog-sidecar/src/service/sidecar_server.rs

```diff
@@ -297,12 +297,12 @@ impl SidecarServer {
 
             let futures = clients
                 .values()
-                .filter_map(|client| client.client.stats().ok())
+                .filter_map(|client| client.client.lock_or_panic().worker.stats().ok())
                 .collect::<Vec<_>>();
 
             let metric_counts = clients
                 .values()
-                .map(|client| client.telemetry_metrics.lock_or_panic().len() as u32)
+                .map(|client| client.client.lock_or_panic().telemetry_metrics.len() as u32)
```

Comment:
you could replace this with a single pass

```rust
let capacity_hint = clients.values().len();
let mut futures = Vec::with_capacity(capacity_hint);
let mut metric_counts = Vec::with_capacity(capacity_hint);

for client in clients.values() {
    let client_guard = client.client.lock_or_panic();
    metric_counts.push(client_guard.telemetry_metrics.len() as u32);
    if let Ok(fut) = client_guard.worker.stats() {
        futures.push(fut);
    }
}
```

Reply:
and this